### PR TITLE
Class methods and respecting user blanklines.

### PR DESCRIPTION
**Pull request by samphippen**

### CODE REVIEW

**Comment by samphippen** [comment]
@georgebrock please provide a review :)

**Comment by samphippen** [comment]
@georgebrock never mind I have found significant bugs here and am reinventing some wheels.

**Comment by samphippen** [comment]
Ok. here's an actual description of what this PR does:

it starts to implement a series of "harder" styling choices including:

* no parens on class method calls, apart from where they are necessary
* breaking arrays and blocks if users expressed them as multiline
* wrapping some constructs to 100 chars
* respecting user blanklines.

@georgebrock please 2 be reviewing

File: src/rubyfmt.rb [hunk 1]
```diff
@@ -4,11 +4,16 @@
 require "pp"
 LineMetadata = Struct.new(:comment_blocks)
 
+# I'm sorry
```
Comment (mikker) [inline src/rubyfmt.rb]: 😆

**Comment by samphippen** [comment]
closing in favour of changes now in master.

**Pull request closed without merging**